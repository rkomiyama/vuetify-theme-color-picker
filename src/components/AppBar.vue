<template>
  <v-app-bar height="64" app>
    <v-app-bar-nav-icon class="hidden-md-and-up" @click.stop="$emit('click:drawer')"></v-app-bar-nav-icon>
    <v-toolbar-title class="headline text-uppercase hidden-xs-only">
      <v-flex class="d-flex align-center">
        <v-img class="hidden-sm-and-down" :src="require('@/assets/github.svg')" height="64" />
        <span>Color&nbsp;</span>
        <span class="font-weight-light">Picker</span>
      </v-flex>
    </v-toolbar-title>
  </v-app-bar>
</template>

<script>
export default {
  name: "AppBar"
};
</script>
