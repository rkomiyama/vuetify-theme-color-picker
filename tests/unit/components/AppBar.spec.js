import Vue from 'vue';
import Vuetify from 'vuetify';
import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer';
import AppBar from '@/components/AppBar.vue'

Vue.use(Vuetify)
let wrapper
const username = "rkomiyama"

describe('AppBar.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(AppBar, {
      propsData: {
        searchUserOption: "username"
      }
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  })

  it('has a search field', () => {
    const searchField = wrapper.find('[data-test="searchFieldTest"]')
    expect(searchField).toBeTruthy()
  })

  it('has a rule for username search', () => {
    const str = 'Invalid username.'
    expect(wrapper.vm.$data.rules.username('rkomiyama')).toBeTruthy()
    expect(wrapper.vm.$data.rules.username('-')).toEqual(str)
    expect(wrapper.vm.$data.rules.username('-r')).toEqual(str)
    expect(wrapper.vm.$data.rules.username('r-')).toEqual(str)
    expect(wrapper.vm.$data.rules.username('-r-')).toEqual(str)
  })

  it('has a rule for fullname search', () => {
    expect(wrapper.vm.$data.rules.fullname()).toBeTruthy()
  })

  it('calls changeUser', () => {
    wrapper.vm.$data.searchField = username
    wrapper.vm.changeUser()
    expect(wrapper.vm.$data.oldSearchField).toEqual(username)
  })

  it('calls turnUserChangeOn', () => {
    wrapper.vm.$data.userChange = false
    wrapper.vm.changeUser()
    expect(wrapper.vm.$data.userChange).toBeTruthy()
  })

  it('calls clearSearchField', () => {
    wrapper.vm.$data.searchField = username
    wrapper.vm.clearSearchField()
    expect(wrapper.vm.$data.searchField).toBeFalsy()
  })

  it('AppBar renders the same HTML', () => {
    const renderer = createRenderer();
    wrapper.vm.$data.rules = {
      username: jest.fn(),
      fullname: jest.fn()
    }
    renderer.renderToString(wrapper.vm, (err, str) => {
    if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
})
