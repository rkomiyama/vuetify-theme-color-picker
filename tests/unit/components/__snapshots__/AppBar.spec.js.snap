// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar.vue AppBar renders the same HTML 1`] = `
<v-app-bar-stub height="64" tag="header" tile="true" extensionHeight="48" src="" value="true" app="true">
  <v-app-bar-nav-icon-stub class="hidden-md-and-up"></v-app-bar-nav-icon-stub>
  <v-toolbar-title-stub class="headline text-uppercase hidden-xs-only">
    <v-flex-stub tag="div" class="d-flex align-center">
      <v-img-stub height="64" position="center center" src="" transition="fade-transition" class="hidden-sm-and-down"></v-img-stub> <span>GitHubÂ </span> <span class="font-weight-light">PERSONAS</span>
    </v-flex-stub>
  </v-toolbar-title-stub>
  <v-spacer-stub class="hidden-xs-only"></v-spacer-stub>
  <v-text-field-stub errorCount="1" errorMessages="" messages="" rules="function () {
        return fn.apply(this, arguments);
      }" successMessages="" value="" backgroundColor="" hideDetails="true" label="Search by Username" prependIcon="mdi-account-search" loaderHeight="2" clearIcon="$vuetify.icons.clear" type="input" maxlength="39" data-test="searchFieldTest"></v-text-field-stub>
</v-app-bar-stub>
`;
